src/transactions/TransactionFrame.cpp

```diff
@@ -286,6 +286,33 @@ TransactionFrame::processSeqNum(LedgerManager& lm, LedgerDelta& delta)
     }
 }
 
+bool
+TransactionFrame::processSignatures(SignatureChecker& signatureChecker,
+                                    Application& app, LedgerDelta& delta)
+{
+    if (app.getLedgerManager().getCurrentLedgerVersion() < 10)
+    {
+        return true;
+    }
+
+    for (auto& op : mOperations)
+    {
+        if (!op->checkSignature(signatureChecker, app, nullptr))
+        {
+            app.getMetrics()
+                .NewMeter({"transaction", "invalid", "invalid-op"},
+                          "transaction")
+                .Mark();
+            markResultFailed();
+            return false;
+        }
+    }
+
+    removeUsedOneTimeSignerKeys(signatureChecker, delta,
```

Comment:
I've implemented this change. It changes one test case to behave as before - https://github.com/stellar/stellar-core/pull/1709/commits/1a9cbd7fbb336943021abd7209b4c9a85ce59a65#diff-a4dc9da0d27fe7b8c7fcbaa5ac1bf72cL455.

Do you think that this behavior is proper one? (one time signature is no longer removed then transaction results in txBAD_AUTH_EXTRA).

And this is similar to checkValid, but it calls checkSignature on operation instead of checkValid. Not sure if this can be easily extracted.